pkg/restapi/issuer/operation/operations.go

```diff
@@ -27,30 +40,91 @@ type Handler interface {
 	Handle() http.HandlerFunc
 }
 
-// New returns issuer instance
-func New() *Operation {
-	svc := &Operation{}
+// Operation defines handlers for authorization service
+type Operation struct {
+	handlers  []Handler
+	cmsConfig *oauth2.Config
+}
+
+// Config defines configuration for issuer operations
+type Config struct {
+	CmsConfig *oauth2.Config
+}
+
+// New returns authorization instance
+func New(config *Config) *Operation {
+	svc := &Operation{cmsConfig: config.CmsConfig}
 	svc.registerHandler()
 
 	return svc
 }
 
-// Operation defines handlers for issuer service
-type Operation struct {
-	handlers []Handler
+// Login using oauth2, will redirect to Auth Code URL
+func (c *Operation) Login(w http.ResponseWriter, r *http.Request) {
+	// AuthCodeURL receives state that is a token to protect the user from CSRF attacks
+	oauthState := generateStateOauthCookie(w)
+
+	u := c.cmsConfig.AuthCodeURL(oauthState)
+	http.Redirect(w, r, u, http.StatusTemporaryRedirect)
+}
+
+// Callback for oauth2 login
+func (c *Operation) Callback(w http.ResponseWriter, r *http.Request) {
+	// read oauthState from cookie
+	oauthState, err := r.Cookie(oauthCookieName)
+	if err != nil {
+		log.Error(err)
+		http.Redirect(w, r, "/", http.StatusTemporaryRedirect)
+
+		return
+	}
+
+	// validate that oauth cookie state matches the the state query parameter on your redirect callback
+	if r.FormValue(stateFormKey) != oauthState.Value {
+		log.Warn("invalid oauth state")
```

Comment:
This probably doesn't provide the event context.

Reply:
No